daphne/src/roles/leader.rs

```diff
@@ -162,7 +162,16 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
             .map_err(|e| DapAbort::from_codec_error(e, task_id.clone()))?;
         debug!("report id is {}", report.report_metadata.id);
 
-        resolve_taskprov(self, task_id, req, Some(&report.report_metadata)).await?;
+        if let Some(taskprov_version) = self.get_global_config().taskprov_version {
```

Comment:
if you are checking this for every use of resolve_taskprov, maybe move the check into resolve_taskprov itself?

Reply:
I thought about that, the reason I kept it out is:
- [in roles/helper.rs](https://github.com/cloudflare/daphne/blob/mendess/cleanup-taskprov/daphne/src/roles/helper.rs#L61-L88) you'd have to check twice
- you can't forget to check because you need the value inside the option to call `resolve_taskprov`